quic/s2n-quic-qns/Cargo.toml

```diff
@@ -24,8 +24,7 @@ http = "1.0"
 humansize = "2"
 lru = "0.10"
 rand = "0.8"
-# dangerous_configuration is used to allow for cert verification to be disabled for the amplification limit interop test
-rustls = { version = "0.21", features = ["dangerous_configuration", "quic"] }
+rustls = "0.23"
```

Comment:
I'd prefer removing this dependency here in favor of accessing it through `s2n_quic::provider::tls::rustls`, just so its fewer references to maintain